### PR TITLE
[Chat] Stabilizing Inline Image and Image Gallery and keep File Sharing in beta

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
# What

Stabilizing Inline Image and Image Gallery
Separate file sharing from inline image
Update storybook and remove beta banners
Update SDK dependencies
Make Enum changes

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/chat-component-bindings/src/messageThreadSelector.ts [hunk 1]
```diff
@@ -349,22 +345,17 @@ export const messageThreadSelectorWithThread: () => MessageThreadSelector = () =
...
-
 const isMessageValidToRender = (message: ChatMessageWithStatus): boolean => {
   if (message.deletedOn) {
     return false;
   }
-  if (
-    message.metadata?.['fileSharingMetadata'] ||
-    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */ message.content?.attachments
-  ) {
+  if (message.metadata?.['fileSharingMetadata'] || message.content?.attachments) {
```
Comment (emlynmac) [inline packages/chat-component-bindings/src/messageThreadSelector.ts:358]: Can we avoid direct access to objects via a string?
Is there a typing on metadata that can be used / extended?

File: packages/communication-react/review/stable/communication-react.api.md [hunk 2]
```diff
@@ -114,6 +115,17 @@ export type AreParamEqual<A extends (props: any) => JSX.Element | undefined, B e
 // @public
 export type AreTypeEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
 
+// @public
+export interface AttachmentDownloadResult {
+    // (undocumented)
```
Comment (emlynmac) [inline packages/communication-react/review/stable/communication-react.api.md:120]: Do we need some documentation on these parameters?

File: packages/communication-react/review/stable/communication-react.api.md [hunk 3]
```diff
@@ -1216,6 +1242,9 @@ export type ChatAdapterUiState = {
     error?: Error;
 };
 
+// @public
+export type ChatAttachmentType = 'file' | 'inlineImage' | 'unknown';
```
Comment (emlynmac) [inline packages/communication-react/review/stable/communication-react.api.md:1246]: Should the 'file' type be exposed to the public API surface?

File: packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx [hunk 4]
```diff
@@ -299,7 +288,6 @@ const MessageBubble = (props: ChatMessageComponentAsMessageBubbleProps): JSX.Ele
     // messageContainerStyle used in className and style prop as style prop can't handle CSS selectors
     chatMessageStyles.body,
     // disable placeholder functionality for GA releases as it might confuse users
```
Comment (JoshuaLai) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:290]: Might want to clean up and remove this comment as well

Comment (JoshuaLai) [inline packages/react-components/src/components/ChatMessage/ChatMessageComponentAsMessageBubble.tsx:290]:
```suggestion
```

File: packages/react-components/review/stable/react-components.api.md [hunk 5]
```diff
@@ -303,6 +312,9 @@ export interface _CaptionsSettingsModalStrings {
     captionsSettingsSpokenLanguageDropdownLabel?: string;
 }
 
+// @public
+export type ChatAttachmentType = 'file' | 'inlineImage' | 'unknown';
```
Comment (JoshuaLai) [inline packages/react-components/review/stable/react-components.api.md:316]: If `file` is still in beta lets not expose that enum type yet

File: packages/communication-react/review/stable/communication-react.api.md (hunk 3 above)
Comment (JoshuaLai) [reply packages/communication-react/review/stable/communication-react.api.md:1246]: If `file` is still in beta lets no expose the enum yet

**Comment by JoshuaLai** [comment]
Under `/workspaces/communication-ui-library/common/config/babel/features.js` I think we can remove the `teams-inline-images-and-file-sharing` feature flag. 
And update` /workspaces/communication-ui-library/packages/react-composites/tests/app/chat/FakeAdapterApp.tsx` to stop using that feature flag and use the `file-sharing` flag instead

**Review by emlynmac — APPROVED** [review]
LGTM!

**Review by JoshuaLai — APPROVED** [review]

**Pull request merged**